### DISCUSSION TITLE
Having Number and Version in Title Page PDF - Quarto book render PDF

### DISCUSSION THREAD

**Discussion by martinlardi** [quarto-dev/quarto-cli#5168] [Q&A]
![image](https://user-images.githubusercontent.com/84940897/231509188-7f5568a0-c9e9-4ae6-b098-b733bfb71b08.png)
![image](https://user-images.githubusercontent.com/84940897/231509247-3b5b671f-f07e-4ee1-abc6-e847efe95484.png)

```
project:
  type: book

book:
  title: "quarto Title"
  subtitle : "this is subtitle"
  number : "Report Number xxx"
  version : "Version 1.0.1"
  author : "person1"
  chapters:
    - index.qmd

format: pdf

editor: visual
```

In the title page on PDF, I want to have "Report Number xxx" and "Version 1.0.1" in between subtitle and author. How do I do that ?

**Comment by rgaiacs** [comment]
Thanks @martinlardi for the question. To add the report number and the version you will have to use partial templates. See the documentation at https://quarto.org/docs/journals/templates.html#latex-partials.

**Comment by martinlardi** [comment]
_quarto.yml
```
project:
  type: book

book:
  title: "quarto Title"
  subtitle : "this is subtitle"
  number : "Report Number xxx"
  version : "Version 1.0.1"
  author : "person1"
  chapters:
    - index.qmd

format:
  pdf:
    template-partials:
      - title.tex

editor: visual
```

title.tex
```
\title{$title$}
\subtitle{$subtitle$}
\number{$number$}
\version{$version$}
\author{$for(author)$$author$$sep$ \and $endfor$}
```

It gave me an error : 
```
compilation failed- error
Missing number, treated as zero.
<to be read again> 
                   {
l.147 \number{
              } 
```


How do I add variable "number" and "version" to title.tex from _quarto.yml ?  am I approaching this incorrectly ?

**Reply by cderv** [reply]
Just for context: You cannot unfortunately create any variable you want under the `book` key. It has a private schema and is protected to avoid potential conflict with any potential value or future value under `book` used by Quarto.  Though it should accept [any value from the CSL schema ](https://github.com/quarto-dev/quarto-cli/issues/2148#issuecomment-1244483385) and `number` and `version` should be part of it. To use them try `$book.version$` and `$book.number$`

See discussion at https://github.com/quarto-dev/quarto-cli/issues/2148 for more about this.
